content/articles/implementation-of-fuzzy-logic-in-matlab/index.md

```diff
@@ -0,0 +1,99 @@
+
+### Fuzzy logic approach
+Suppose we poll 100 bankers and ask them to assign the transition from bad to neutral credit and change from neutral to good. More than likely, in this case, there are going to be a different value for this. Let us assume that every banker thought 750 and above is good credit, but only half thought 700 was good, and none thought 650 was good. It traces out the particular membership function for good credit. We can do the same for neutral credit and bad credit. Here, we will end up with three overlapping membership functions.
+
+![fuzzy logic](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_two.png)
+
+Now, we can see that the statement's truth does not belong to a single set but can belong to multiple sets based on the degree of the truth.
+
+### Implementation of the fuzzy logic in Matlab
+Consider a very simple example;
+We need to control the speed of a motor by changing the input voltage when a set point is defined. If the motor runs fast for some reason, we need to slow it down by reducing the input voltage. Conversely, if the motor slows below the set point, the input voltage must be increased so that the motor reaches the set point.
+let the input status word be;
+Too slow
+Just right
+Too fast
+
+let the output action words be;
+Less voltage(slow down)
+No change
+More voltage(speed up)
+
+The rules base are;
+1. if the motor is running too slow, then more voltage.
+2. If the motor speed is right, then no change.
+3. If the motor speed is too fast, then the voltage is less.
+
+We will define all these in a fuzzy system and create a file for this. The first step is to open the fuzzy logic toolbox. To do this, execute the `fuzzy` command in the command window. When you open the logic toolbox, a new window shown below will pop up.
+
+![fuzzy toolbox](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_three.jpeg)
+
+Here, we have three blocks, is, input, Mamdani and output. In the input, this is where we input our data and for the fuzzification process. The output section, which displays the output results, acts as a defuzzification. Finally, the Mamdani is for the rule-based block. What it means is that you define your rules in this box. Now double-click on the input to open it up. When you do so, a window shown below opens up.
+
+![input window](/engineering-education/Implementation-of-fuzzy-logic-in-matlab/fuzzy_four.jpeg)
+*figure1.0*
+
+Here, we have three inputs `mf1`, `mf2` and `mf3`. We should now change these names according to our inputs so that they can make sense. Note that you do the renaming independently. To do it, click on the box with the current name, delete it, replace it with the name you want it to be, and click `enter`. For example, changing our `mf1` to `slow`, and the output is below.
```

Comment:
Always use good and understandable variable names

Reply:
The `mf1` and such are variable names that are already in matlab but we change them as explained in the tutorial.

Reply:
@collince777 Changing them would look good